dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/Serializer.java

```diff
@@ -55,14 +63,23 @@
     return builder;
   }
 
-  static List<Dimension> toListOfDimensions(Labels labels) {
-    ArrayList<Dimension> dimensions = new ArrayList<>(labels.size());
-    labels.forEach((k, v) -> dimensions.add(Dimension.create(k, v)));
+  static List<Dimension> toListOfDimensions(Attributes attributes) {
+    ArrayList<Dimension> dimensions = new ArrayList<>(attributes.size());
+
+    attributes.forEach(
+        (k, v) -> {
+          if (k.getType() == AttributeType.STRING) {
+            dimensions.add(Dimension.create(k.getKey(), String.valueOf(v)));
+          } else {
+            logger.warning(String.format(TEMPLATE_MSG_NON_SUPPORTED_ATTRIBUTE_TYPE, k.getType()));
```

Comment:
This will always invoke `String.format`, regardless of log-level. I have seen this in a few other places as well which were not part of your PR. 

I would suggest using [this overload of warning](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html#warning-java.util.function.Supplier-) which takes a supplier, or [this overload of `log`](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html#log-java.util.logging.Level-java.lang.String-java.lang.Object-).

Reply:
Cool! I changed to use the "lazy" overload everywhere.